### ISSUE TITLE
How to include html in campfire lines?

### ISSUE THREAD

**Issue by goldhat** [basecamp/bc3-api#182]
When I use html in campfire line posts such as { content: "<h1>Message Header</h1><br />message body..."} the html is displayed as written in the campfire chat, and not parsed as html. Is this some type of encoding/escaping issue? I'm setting the message content in an array and using json_encode. No other slashing or encoding at this point.

**Comment by georgeclaghorn** [comment]
I’m afraid that HTML chat lines can only be posted by interactive chatbots in response to a command. We only accept plain-text chat lines via the API. Sorry about that!

**Comment by goldhat** [comment]
Thanks for the fast reply. We do plan to use a chatbot (testing right now without chatbot) so your reply helps, but can you clarify "in response to a command", is that different from creating a chatbot and just posting lines (docs suggests line posts would be plain text). Would that be using the command_url option mentioned at https://github.com/basecamp/bc3-api/blob/master/sections/chatbots.md#create-a-chatbot ?

I'm not clear on what triggers use of the command_url. What we're trying to copy is how github posts commit notifications to campfire (which has html, links etc).

**Comment by georgeclaghorn** [comment]
When a chatbot has a `command_url`, it can receive commands in Campfire. For example, if I add a chatbot named Yo with a `command_url` of `https://yo.example.com/messages`, posting the following in a Campfire will trigger a POST request to that URL:

```
!yo Joel
```

The request body will look something like this:

```json
{
  "command": "What up?",
  "creator": {
    "id": 1007299143,
    "attachable_sgid": "BAh7CEkiCGdpZAY6BkVUSSIrZ2lkOi8vYmMzL1BlcnNvbQQcMDA3Mjk5MTQzP2V4cGlyZXNfaW4GOwBUSSIMcHVycG9zZQY7AFRJIg9hdHRhY2hhYmxlBjsAVEkiD2V4cGlyZXNfYXQGOwBUMA==--919d2c8b11ff403eefcab9db42dd26846d0c3102",
    "name": "Victor Cooper",
    "email_address": "victor@honchodesign.com",
    "personable_type": "User",
    "title": "Chief Strategist",
    "bio": "Don't let your dreams be dreams",
    "created_at": "2016-09-22T16:21:03.625-05:00",
    "updated_at": "2016-09-22T16:21:06.184-05:00",
    "admin": true,
    "owner": true,
    "time_zone": "America/Chicago",
    "avatar_url": "https://3.basecamp-static.com/195539477/people/BAhpBEcqCjw=--c632b967cec296b87363a697a67a87f9cc1e5b45/avatar-64-x4",
    "company": {
      "id": 1033447817,
      "name": "Honcho Design"
    }
  },
  "callback_url": "https://3.basecamp.com/195539477/integrations/2uH9aHLEVhhaXKPaqrj8yw8P/buckets/2085958501/chats/9007199254741775/lines"
}
```

The chatbot can respond to the POST request from Basecamp with HTML, which will be added to the Campfire as a line. That’s what I was referring to when I said “in response to a command.”

Anyway, I just discovered that the docs for [creating a line as a chatbot](https://github.com/basecamp/bc3-api/blob/master/sections/chatbots.md#create-a-line) are incorrect. They indicate that a plain text body is accepted, but HTML is also allowed there. I’ll update them.

In summary, chatbots can post limited HTML—regardless of whether they do so in response to a command, contrary to my previous comment. Other people can only post plain text chat lines via the API.
